crates/rpc/rpc/src/trace.rs

```diff
@@ -38,6 +38,7 @@ use tokio::sync::{AcquireError, OwnedSemaphorePermit};
 /// This type provides the functionality for handling `trace` related requests.
 pub struct TraceApi<Eth> {
     inner: Arc<TraceApiInner<Eth>>,
+    eth_config: EthConfig,
```

Comment:
we also want the settings in the inner type, that's what the inner type is for that that cloning TraceApi per request is as cheap as possible